virtcontainers/qemu.go

```diff
@@ -1567,6 +1587,32 @@ func (q *qemu) resizeVCPUs(reqVCPUs uint32) (currentVCPUs uint32, newVCPUs uint3
 	return currentVCPUs, newVCPUs, nil
 }
 
+func (q *qemu) updateMemoryBalloon(sizeMB uint32) error {
+
+	if q.state.HasVFIODevice {
+		// When VFIO devices are used we dont want to get pages back to
+		// the host. Lets increase the balloon to a large amount of
+		// memory:
+		// - The guest memmory is not affected
+
+		// TODO: This is not required for qemu +v3.1.0 where "Inhibit
+		// ballooning" is implemented.
+
+		// Reference:
+		// https://www.ibm.com/support/knowledgecenter/en/SSZJY4_3.1.0/liabp/liabpmemoryballooning.htm
+		// https://bugs.launchpad.net/qemu/+bug/1762707
+		hostMB, err := q.hostMemMB()
+		if err != nil {
+			return err
+		}
+		sizeMB = uint32(hostMB)
+		q.Logger().WithField("memory-mb", sizeMB).Warnf("VM uses VFIO, balloon size will be total host memory")
```

Comment:
You also mentioned that the `Inhibit Balooning` feature is available in 3.1.0.
Considering this plus various other features, we should really be updating qemu version that we package.
cc @egernst

Reply:
@amshinde  yeah this sounds a more safe approach, we can set as pre-requisite before we land this PR.